[FIX] hw_drivers: improve image update reliability

loouis-t opened the pull request:
We improve the image reliability by:

- Improving handlers cleanup (we only removed os-specific handlers but downloaded all),
- Forcing Odoo service to restart after checkout, even if an error occurred.

Task: 4433461

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/191928.png)](https://mergebot.odoo.com/odoo/odoo/pull/191928)

loouis-t commented on addons/hw_drivers/tools/helpers.py:
```diff
@@ -400,10 +404,10 @@ def delete_iot_handlers():
-                unlink_file(f"odoo/addons/hw_drivers/iot_handlers/{directory}/{file}")
+            iot_handlers = file_path(f'hw_drivers/iot_handlers/{directory}')
+            for file in Path(iot_handlers).iterdir():
+                if file.is_file():
```
`__pycache__` directories can't be deleted by `unlink_file()` method

loouis-t commented:
@odoo/rd-security could you have a look at this please?

## Context

We encountered permission errors while downloading IoT handlers (since we switched from `pi` user for Odoo service to `odoo` user).
We also had an issue while checking out the code on the IoT Box: if a bash command failed, we did not restart the service.

### Solution proposed

We fixed the deletion of handlers and ensured Odoo service restart after checking out.

## Security issues

- `subprocess.run(`
  - **Safe because:** we only re indented this part, we did not touch the logic.
- `for file in Path(iot_handlers).iterdir():`
  - **Reason for the use:** We need to delete all files in the directory (either `iot_handlers/drivers` or `iot_handlers/interfaces`),
  - **Safe because:** path is hardcoded.

Ysoroko reviewed (approved):
@robodoo r+

Ysoroko commented:
@robodoo r+
